src/util.js

```diff
@@ -106,3 +109,86 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function container_stop(container, timeout) {
+    const id = container.ID;
+    if (timeout == undefined) {
+        timeout = 10;
+    }
+    varlinkCall(PODMAN, "io.podman.StopContainer", {name: id, timeout: timeout})
+        .then(reply => {
+            console.log(reply.container);
+            return reply.container;
+        })
+        .catch(ex => console.error("Failed to do RemoveContainerForce call:", JSON.stringify(ex)));
+}
+
+export function updateContainers() {
+    let newContainers = [];
+    let newContainersStats = [];
+    return new Promise((resolve, reject)=> {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                    .then(reply => {
+                        let newContainersMeta = reply.containers;
+                        if (!newContainersMeta) {
+                            resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                            return;
+                        }
+                        const len = newContainersMeta.length;
+                        newContainersMeta.map((container) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                .then(reply => {
+                                    newContainers.push(JSON.parse(reply.container));
+                                })
+                                .catch(ex => console.error("Failed to do InspectContainer call:", ex, JSON.stringify(ex)));
+                        });
+                        newContainersMeta.map((container) => {
+                            varlinkCall(PODMAN, "io.podman.GetContainerStats", {name: container.id})
+                                .then(reply => {
+                                    if (reply.container) {
+                                        newContainersStats[container.id] = reply.container;
+                                        console.log(Object.keys(newContainersStats).length);
+                                    }
+                                    if (newContainers.length === len && Object.keys(newContainersStats).length === len) {
+                                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                                    }
```

Comment:
This is brittle. If any single one of the `InspectContainer()` or `GetContainerStats()` calls fails, the promise will never be resolved. Instead, collect the promises of all calls in an array, and after this `newContainersMeta` loop use [Promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) to  pick up "everything succeeded" vs. "something failed".